driver/amr.c

```diff
@@ -11,6 +11,146 @@ static void usage(const char *exe_name) {
   fprintf(stderr, "%s <input.yaml>\n\n", exe_name);
 }
 
+typedef struct {
+  char dir[PETSC_MAX_PATH_LEN];
+  char file[PETSC_MAX_PATH_LEN];
+
+  PetscBool is_defined;          // whether to refine or not
+  struct tm date, current_date;  // date for refinement
+} RefinementDataset;
+
+static PetscErrorCode DetermineDatasetFilename(struct tm *current_date, char *dir, char *file) {
+  PetscFunctionBegin;
+
+  mktime(current_date);
+  snprintf(file, PETSC_MAX_PATH_LEN - 1, "%s/%4d-%02d-%02d:%02d-%02d.%s.bin", dir, current_date->tm_year + 1900, current_date->tm_mon + 1,
+           current_date->tm_mday, current_date->tm_hour, current_date->tm_min, PETSC_ID_TYPE);
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+#define NUM_DATE_VALUES 5  // number of parameters expected for date
+PetscErrorCode ParseRefinementDataOptions(RefinementDataset *refinement_dataset) {
+  PetscFunctionBegin;
+
+  // set default settings
+  refinement_dataset->is_defined = PETSC_FALSE;
+
+  // parse command line options
+  PetscInt date[NUM_DATE_VALUES];
+  PetscInt ndate = NUM_DATE_VALUES;
+
+  PetscBool refine_flag;
+  PetscCall(PetscOptionsGetIntArray(NULL, NULL, "-refine_data_start_date", date, &ndate, &refine_flag));
```

Comment:
Do you think we'll eventually want these options in some AMR-related input section?

Reply:
At this stage, it is unclear whether we will conduct numerous simulations using AMR, apart from an initial simulation.